tests/admin/test_controller.py

```diff
@@ -1215,6 +1216,291 @@ def test_custom_list_delete_errors(self):
+            eq_(LANE_WITH_PARENT_AND_DISPLAY_NAME_ALREADY_EXISTS, response)
+
+        with self.request_context_with_library("/", method="POST"):
+            flask.request.form = MultiDict([
+                ("display_name", "lane2"),
+                ("custom_list_ids", json.dumps([list.id])),
+            ])
+            response = self.manager.admin_lanes_controller.lanes()
+            eq_(LANE_WITH_PARENT_AND_DISPLAY_NAME_ALREADY_EXISTS, response)
+
+        with self.request_context_with_library("/", method="POST"):
+            flask.request.form = MultiDict([
+                ("parent_id", "12345"),
+                ("display_name", "lane"),
+                ("custom_list_ids", json.dumps([list.id])),
+            ])
+            response = self.manager.admin_lanes_controller.lanes()
+            eq_(MISSING_LANE.uri, response.uri)
+
+        with self.request_context_with_library("/", method="POST"):
+            flask.request.form = MultiDict([
+                ("parent_id", lane1.id),
+                ("display_name", "lane"),
+                ("custom_list_ids", json.dumps(["12345"])),
+            ])
+            response = self.manager.admin_lanes_controller.lanes()
+            eq_(MISSING_CUSTOM_LIST.uri, response.uri)
+
+    def test_lanes_create(self):
+        data_source = DataSource.lookup(self._db, DataSource.LIBRARY_STAFF)
+        list, ignore = create(self._db, CustomList, name=self._str, data_source=data_source)
+        list.library = self._default_library
+
+        # The new lane's parent has a sublane already.
+        parent = self._lane("parent")
+        sibling = self._lane("sibling", parent=parent)
+
+        with self.request_context_with_library("/", method="POST"):
+            flask.request.form = MultiDict([
+                ("parent_id", parent.id),
+                ("display_name", "lane"),
+                ("custom_list_ids", json.dumps([list.id])),
+            ])
+            response = self.manager.admin_lanes_controller.lanes()
+            eq_(201, response.status_code)
+
+            [lane] = self._db.query(Lane).filter(Lane.display_name=="lane")
+            eq_(lane.id, int(response.response[0]))
+            eq_(self._default_library, lane.library)
+            eq_("lane", lane.display_name)
+            eq_(parent, lane.parent)
+            eq_(1, len(lane.customlists))
+            eq_(list, lane.customlists[0])
+            eq_(0, lane.priority)
+
+            # The sibling's priority has been shifted down to put the new lane at the top.
+            eq_(1, sibling.priority)
+
+    def test_lanes_edit(self):
+        data_source = DataSource.lookup(self._db, DataSource.LIBRARY_STAFF)
+        list1, ignore = create(self._db, CustomList, name=self._str, data_source=data_source)
+        list1.library = self._default_library
+        list2, ignore = create(self._db, CustomList, name=self._str, data_source=data_source)
+        list2.library = self._default_library
```

Comment:
You do this a lot and I think you could use `self._customlist(data_source_name=DataSource.LIBRARY_STAFF, num_entries=0)` isntead.